src/main/java/com/ldtteam/structurize/placement/StructurePlacer.java

```diff
@@ -399,7 +399,7 @@ public BlockPlacementResult getResourceRequirements(
     {
         final BlockState worldState = world.getBlockState(worldPos);
         boolean sameBlockInWorld = false;
-        if (worldState.getBlock() == localState.getBlock())
+        if (worldState.getBlock() == localState.getBlock() && tileEntityData != null)
```

Comment:
Did you mean `== null` here?  Otherwise this looks like it will be `false` a lot more than it should, e.g. for non-DO blocks.

Reply:
yes, u are right